rclcpp/test/rclcpp/test_duration.cpp

```diff
@@ -148,6 +148,34 @@ TEST_F(TestDuration, from_seconds) {
     rclcpp::Duration::from_seconds(-1.5));
 }
 
+TEST_F(TestDuration, from_rmw_time) {
+  constexpr auto max_rcl_duration = std::numeric_limits<rcl_duration_value_t>::max();
+  {
+    rmw_time_t rmw_duration{};
+    rmw_duration.sec = max_rcl_duration + 1uLL;
```

Comment:
Seems like there is a value of `rmw_duration.sec` that is less than `max_rcl_duration` which will still overflow the duration. This is related to my other comment. Like `rmw_duration.sec = max_rcl_duration / 10uLL;` will likely cause the problem because `RCL_NS_TO_S` will multiply it by much more.

Reply:
https://github.com/ros2/rclcpp/pull/1467/commits/655f76e14c95876951e268c7c3d63bc185ede1a1 should cover this case correctly now.